### DISCUSSION TITLE
ドキュメント改善:コンポーネントにAvatar Optimizerとの互換性をもたせる

### DISCUSSION THREAD

**Discussion by whiteflare** [anatawa12/AvatarOptimizer#1142] [General]
https://vpm.anatawa12.com/avatar-optimizer/ja/docs/developers/make-your-components-compatible-with-aao/

現在の記載について、追加した方が良い記載があると考えています。

1. ComponentInformation はAAO内部では Renderer や PhysBone といったランタイムで動作するコンポーネントを登録するために使用している
2. ComponentInformation を用いると独自コンポーネントを作成したときにAAOに登録することができる
3. NDMFプラグインを作成したとき、可能であればAAO処理前に処理をしてほしい (AAOとしては全ての編集処理が完了した後に最適化を行いたいため)

この情報は、remove-component / register-component よりも前に記載されていると良いと考えています。
remove-component / register-component は実際の対応手順を記したものであるため、意思決定 (独自コンポーネント作者にとっては削除するか登録するか) は対応手順よりも前に行われているべきであり、対応手順内に記載されていると混乱するだけになってしまうと思います。

その他、細かい話としては、

> 副作用のあるコンポーネントである。

副作用という単語は制作者ごとにイメージが分かれるので、可能ならば避けた方が良い表現です。

> コンポーネントを削除する方法はいくつかあります。

削除は DestroyImmediate を指すと明記したほうが良いと思います。

> // call methods on the collector to tell about the component

CollectDependency という名称的には「独自コンポーネントからの依存を登録する」と捉えられると思います。
実際には独自コンポーネント自身の情報も登録してほしいため、XMLDOCにはその旨を記載するべきです。

**Comment by anatawa12** [comment]
#1143 で改善を図ったので確認していただきたいです

**Reply by whiteflare** [reply]
en側を見れていないですが、ja側は見ました。かなり伝わりやすくなったと思います。

ツール制作者視点でのAssetDescriptionの立ち位置もわかりやすくなりましたね。AssetDescriptionは専用ページがあるので「下のセクション」が無くても良いかなぁ。
ベストは「Asset Descriptionを作成する」という見出しと「AssetDescription を参照してください。」の本文がある状態です。そのようにしてあれば、見出しだけ見て状況が伝わります。
![image](https://github.com/user-attachments/assets/d5d1dedd-6b8e-4f97-9505-927c3b757120)

ぜんぜん関係ないところですが確認中にAsset Descriptionについてひとつ見つけてしまいまして、
https://github.com/anatawa12/AvatarOptimizer/blob/629b553122a0a543ca8b308996f9392e165e5787/Editor/AssetDescription.cs#L73
は
```
baseUrl += "docs/developers/asset-description/";
```
が正しいと思います。ボタンの先が404になっていました。

**Reply by anatawa12** [reply]
> ベストは「Asset Descriptionを作成する」という見出しと「AssetDescription を参照してください。」の本文がある状態です。そのようにしてあれば、見出しだけ見て状況が伝わります。

見出しについては完全に考慮を考えてませんでした。追加しておきます。
